hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java

```diff
@@ -196,8 +196,19 @@ private EntityManagerFactoryBuilderImpl(
 			integrationSettings = Collections.emptyMap();
 		}
 
+		Map mergedIntegrationSettings = null;
+		Properties properties = persistenceUnit.getProperties();
+		if ( properties != null ) {
+			mergedIntegrationSettings = new HashMap( integrationSettings );
+			mergedIntegrationSettings.putAll( persistenceUnit.getProperties() );
+		}
+
```

Comment:
I do not think it is so simple as adding to a map the `integrationSettings` and then `persistenceUnit.getProperties()`, see `MergedSettings` creation at https://github.com/hibernate/hibernate-orm/pull/3227/files#diff-c1662f25d34183e3651ada18824bbd27R216

Reply:
@dreab8 The goal of this change is to allow the Persistence Unit properties to be passed to the `BootstrapServiceRegistry` creation process.

The `MergedSettings` object requires the `BootstrapServiceRegistry` instance, hence this change. So, you cannot create a `MergedSettings` without a `BootstrapServiceRegistry`. 

But the goal of this PR is not to provide all possible settings to the `BootstrapServiceRegistry`, as the  `MergedSettings does. You only need the JPA properties as they are the typical way of configuring a JPA application.

Also, this change does not impact how the `MergedSettings` is used. It just provides more info to the `BootstrapServiceRegistry`, and the temporary `mergedIntegrationSettings` is never used after the `BootstrapServiceRegistry` instance gets created.

All in all, I think this is safe to be integrated, having a test case that proves the fix, and all other tests have run successfully too.

Reply:
I didn;t want to say that you have to use `MergedSettings` but that it is an example showing that the logic is not so easy in case of values that are presents in both `integrationSettings` and `persistenceUnit.getProperties()` .

In your solution the properties of `PersistenceUnitDescriptor` override the `integrationSettings` ones and I think this is not correct.

Reply:
That's a good observation. I could initialize the new `Map` using the `persistenceUnit.getProperties()` and call `putAll` using the `integrationSettings`, as these are provided at bootstrap time, and should take precedence.

Should I refactor it like this?

Reply:
I rebased the changes according to your suggestion.

Reply:
Awesome!

Reply:
thanks @vladmihalcea

Reply:
The more complex merging that Andrea is talking about (I think) is more about, e.g., JDBC config from integrationSettings ought to override not just JDBC config from PU properties - it should also override datasource config from the PU properties.  I can't say whether those more complete overriding is needed here - probably not